MediaBrowser.WebDashboard/Api/DashboardService.cs

```diff
@@ -111,24 +114,41 @@ public class DashboardService : IService, IRequiresRequest
         /// </summary>
         /// <param name="logger">The logger.</param>
         /// <param name="appHost">The application host.</param>
+        /// <param name="appConfig">The application configuration.</param>
         /// <param name="resourceFileManager">The resource file manager.</param>
         /// <param name="serverConfigurationManager">The server configuration manager.</param>
         /// <param name="fileSystem">The file system.</param>
         /// <param name="resultFactory">The result factory.</param>
         public DashboardService(
             ILogger<DashboardService> logger,
             IServerApplicationHost appHost,
+            IConfiguration appConfig,
             IResourceFileManager resourceFileManager,
             IServerConfigurationManager serverConfigurationManager,
             IFileSystem fileSystem,
             IHttpResultFactory resultFactory)
         {
             _logger = logger;
             _appHost = appHost;
+            _appConfig = appConfig;
             _resourceFileManager = resourceFileManager;
             _serverConfigurationManager = serverConfigurationManager;
             _fileSystem = fileSystem;
             _resultFactory = resultFactory;
+
+            // If hosting the web client, validate the client content path
+            if (appConfig.HostWebClient())
+            {
+                string webContentPath = DashboardUIPath;
+                if (!Directory.Exists(webContentPath) || Directory.GetFiles(webContentPath).Length == 0)
```

Comment:
isn't it now done twice?

Reply:
Yes, thank you! I moved this check to `Program.cs` and forgot to remove this instance. Fixed now